From e41630841b6041a4525172263d38f6ec59221cda Mon Sep 17 00:00:00 2001
From: Liu Wenlong <liuwl.fnst@cn.fujitsu.com>
Date: Fri, 14 Jul 2017 14:00:38 -0400
Subject: [PATCH] rpc.rusersd/rup.c: fix function "clnt_call" timeout issue.

parameter "timeout" in function "clnt_call" call was 0 and
lead to an error "RPC: Timed out" when users try to run command "rup hostname".

So, add 25s timeout for function "clnt_call" call in "rup.c",
which is the same with "clnt_call" call in "rusers.c".

Signed-off-by: Liu Wenlong <liuwl.fnst@cn.fujitsu.com>
---
 ...pc.rusersd-rup-fix-clntcall-timeout-issue.patch | 29 ++++++++++++++++++++++
 .../netkit-rusers/netkit-rusers_0.17.bb            |  1 +
 2 files changed, 30 insertions(+)
 create mode 100644 meta-networking/recipes-netkit/netkit-rusers/netkit-rusers/rpc.rusersd-rup-fix-clntcall-timeout-issue.patch

diff --git a/meta-networking/recipes-netkit/netkit-rusers/netkit-rusers/rpc.rusersd-rup-fix-clntcall-timeout-issue.patch b/meta-networking/recipes-netkit/netkit-rusers/netkit-rusers/rpc.rusersd-rup-fix-clntcall-timeout-issue.patch
new file mode 100644
index 0000000..22d128d
--- /dev/null
+++ b/meta-networking/recipes-netkit/netkit-rusers/netkit-rusers/rpc.rusersd-rup-fix-clntcall-timeout-issue.patch
@@ -0,0 +1,29 @@
+diff -uprN netkit-rusers-0.17.orig/rup/rup.c netkit-rusers-0.17/rup/rup.c
+--- netkit-rusers-0.17.orig/rup/rup.c	2017-07-14 13:46:21.030808408 -0400
++++ netkit-rusers-0.17/rup/rup.c	2017-07-14 13:49:01.306808408 -0400
+@@ -58,6 +58,8 @@ static int print_rup_data(const char *ho
+ 
+ static int printtime;			/* print the remote host(s)'s time */
+ static int printseconds;                /* print in seconds, not formatted */
++static struct timeval timeout = { 25, 0 };
++					/* timeout for function "clnt_call"*/
+ 
+ struct host_list {
+ 	struct host_list *next;
+@@ -245,7 +247,6 @@ onehost(const char *host)
+ {
+ 	CLIENT *rstat_clnt;
+ 	statstime host_stat;
+-	struct timeval foo;
+ 	
+ 	rstat_clnt = clnt_create(host, RSTATPROG, RSTATVERS_TIME, "udp");
+ 	if (rstat_clnt == NULL) {
+@@ -257,7 +258,7 @@ onehost(const char *host)
+ 	if (clnt_call(rstat_clnt, RSTATPROC_STATS, 
+ 		      (xdrproc_t) xdr_void, NULL, 
+ 		      (xdrproc_t) xdr_statstime, (char *)&host_stat, 
+-		      foo) != RPC_SUCCESS) 
++		      timeout) != RPC_SUCCESS) 
+ 	{
+ 		warnx("%s",  clnt_sperror(rstat_clnt, host));
+ 		return;
diff --git a/meta-networking/recipes-netkit/netkit-rusers/netkit-rusers_0.17.bb b/meta-networking/recipes-netkit/netkit-rusers/netkit-rusers_0.17.bb
index 459e39e..36175cb 100644
--- a/meta-networking/recipes-netkit/netkit-rusers/netkit-rusers_0.17.bb
+++ b/meta-networking/recipes-netkit/netkit-rusers/netkit-rusers_0.17.bb
@@ -9,6 +9,7 @@ DEPENDS = " tcp-wrappers libtirpc rpcbind"
 SRC_URI = "http://http.debian.net/debian/pool/main/n/${BPN}/${BPN}_${PV}.orig.tar.gz;name=archive \
            http://http.debian.net/debian/pool/main/n/${BPN}/${BPN}_${PV}-8.diff.gz;name=patch8 \
            file://rpc.rusersd-Makefile-fix-parallel-build-issue.patch \
+           file://rpc.rusersd-rup-fix-clntcall-timeout-issue.patch \
 "
 
 SRC_URI[archive.md5sum] = "dc99a80b9fde2ab427c874f88f1c1602"
-- 
2.7.4

